jit/compileoperator.cpp

```diff
@@ -0,0 +1,92 @@
+#include "compileoperator.h"
+#include "llvm/IR/Verifier.h"
+#include "rowgroup.h"
+
+namespace execplan
+{
+// TODO: change row to llvm param
+void compileOperator(llvm::Module& module, const execplan::SRCP& expression, rowgroup::Row& row)
+{
+  auto columns = expression.get()->simpleColumnList();
+  llvm::IRBuilder<> b(module.getContext());
+  llvm::Type* return_type;
+  auto* data_type = b.getInt8Ty()->getPointerTo();
+  auto* isNull_type = b.getInt1Ty()->getPointerTo();
+  switch (expression->resultType().colDataType)
+  {
+    case execplan::CalpontSystemCatalog::BIGINT:
+    case execplan::CalpontSystemCatalog::INT:
+    case execplan::CalpontSystemCatalog::MEDINT:
+    case execplan::CalpontSystemCatalog::SMALLINT:
+    case execplan::CalpontSystemCatalog::TINYINT:
+    case CalpontSystemCatalog::UBIGINT:
+    case CalpontSystemCatalog::UINT:
+    case CalpontSystemCatalog::UMEDINT:
+    case CalpontSystemCatalog::USMALLINT:
+    case CalpontSystemCatalog::UTINYINT:
+      return_type = b.getInt64Ty(); break;
+    case execplan::CalpontSystemCatalog::DOUBLE:
+    case execplan::CalpontSystemCatalog::UDOUBLE: return_type = b.getDoubleTy(); break;
+    case execplan::CalpontSystemCatalog::FLOAT:
+    case execplan::CalpontSystemCatalog::UFLOAT: return_type = b.getFloatTy(); break;
+    default: throw logic_error("compileOperator: unsupported type");
+  }
+  auto* func_type = llvm::FunctionType::get(return_type, {data_type, isNull_type}, false);
+  auto* func =
+      llvm::Function::Create(func_type, llvm::Function::ExternalLinkage, expression->alias(), module);
```

Comment:
I would suggest
```c++
func->setDoesNotThrow();
```
so that no `.eh_frame` or `.rela.eh_frame` section will be generated.

Reply:
Done.